packages/firestore/src/util/misc.ts

```diff
@@ -28,8 +29,16 @@ export class AutoId {
     const chars =
       'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
     let autoId = '';
-    for (let i = 0; i < 20; i++) {
-      autoId += chars.charAt(Math.floor(Math.random() * chars.length));
+    while (autoId.length < 20) {
+      const bytes = PlatformSupport.getPlatform().randomBytes(40);
+      bytes.forEach(b => {
```

Comment:
You could rephrase this with a single for loop:

```ts
const numBytes = 40;
const maxValue = 62 * 4;
let bytes = PlatformSupport.getPlatform().randomBytes(numBytes);
let i = 0;
while (autoId.length < 20) {
  if (i >= numBytes) {
    bytes = PlatformSupport.getPlatform().randomBytes(numBytes);
    i = 0;
  }
  const b = bytes[i++];
  // Avoid modulus bias by discarding byte values greater than 247.
  if (b < maxValue) {
    autoId += chars.charAt(b % 62)
  }
}
```

Not sure this is necessarily better.

Reply:
Right..it's hard to say which one is better. I'll stick to my version..